File: Leetcode_POTD_2025/8_August/3477_Fruits_Into_Baskets_II.cpp
class Solution {
public:
    int numOfUnplacedFruits(vector<int>& fruits, vector<int>& baskets) {
        int r_c = 0;
        vector<bool> p_b(baskets.size(), false);
        for (int q_f : fruits) {
            bool d_p = false;
            for (int i = 0; i < baskets.size(); ++i) {
                if (!p_b[i] && baskets[i] >= q_f) {
                    p_b[i] = true;
                    d_p = true;
                    break;
                }
            }
            if (!d_p) {
                r_c++;
            }
        }
        return r_c;
    }
};

File: Leetcode_POTD_2025/8_August/3477_Fruits_Into_Baskets_II.java
class Solution {
    public int numOfUnplacedFruits(int[] fruits, int[] baskets) {
        int t_u_f = 0;
        boolean[] u_b = new boolean[baskets.length];
        for (int c_f : fruits) {
            boolean f_f = false;
            for (int i = 0; i < baskets.length; ++i) {
                if (!u_b[i] && baskets[i] >= c_f) {
                    u_b[i] = true;
                    f_f = true;
                    break;
                }
            }
            if (!f_f) {
                t_u_f++;
            }
        }
        return t_u_f;
    }
}
